packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
I only mention it because to me, `add` makes the most sense for adding two `Array val`, where I can't append.  In general, `add` seems most useful for immutables.  I think this is why the signature for `String` uses `box + box => val`.

Intuitively, I'd want this signature to be the same: `box + box = val`.  However, there is a difference at play: `String` is always a `Seq` of `U8 val`, but `Array` may be an array of non-`val`, in which case it's impossible to create a `val` array from them.  This is obviously where viewpoint transformation comes into play, but without playing with it myself I'm not sure exactly what to do.

As a method for confirming your approach is correct, be sure to try different caps in your tests, including the cap of `Array` and the cap of `A`.  Make sure that the resulting cap makes sense in each case.